contracts/WildcardSteward_v3.sol

```diff
@@ -353,13 +353,39 @@ contract WildcardSteward_v3 is Initializable {
     function changeReceivingBenefactor(
         uint256 tokenId,
         address payable _newReceivingBenefactor
-    ) public onlyReceivingBenefactorOrAdmin(tokenId) {
+    )
+        public
+        onlyReceivingBenefactorOrAdmin(tokenId)
+        updateBenefactorBalance(benefactors[tokenId])
+        updateBenefactorBalance(_newReceivingBenefactor)
+    {
         address oldBenfactor = benefactors[tokenId];
-        require(
-            oldBenfactor != _newReceivingBenefactor,
-            "Cannot change to same address"
-        );
+
+        require(oldBenfactor != _newReceivingBenefactor);
+        require(address(0) != _newReceivingBenefactor);
```

Comment:
Extra check I added, makes sense to check we not setting this to zero by some weird default.